Fixes for searchDialog, RegexHelperDialog, exception logging

AndisGrossteins opened the pull request:
#### PR Classification
Code cleanup and modernization. Should fix #476.

#### PR Summary
This pull request refactors and modernizes the codebase for improved readability, maintainability, and robustness, while enhancing logging and UI consistency.
- **RegexHelperDialog.cs**: Replaced `comboBox` item management with encapsulated history lists for better data handling.
- **SearchDialog.Designer.cs**: Refactored UI code to remove redundant `this` references, modernized control properties, and streamlined form setup. (mostly automatic shanges by Visual Studio designer mode)
- **Program.cs**: Improved exception logging to log exception instead of event args.
- **PluginRegistry.cs**: Moved logging into if statement to reduce erroneous log messages.

AndisGrossteins committed:
Fix text Regex Helper button

Set SearchDialog.buttonRegexHelper.Autosize

AndisGrossteins committed:
Fix exception in RegexHelperDialog

Insert new items into comboBox dataSource - _expressionHistoryList and _testtextHistoryList instead of directly into comboBox.Items.

AndisGrossteins committed:
Log actual exception instead of ThreapdExceptionEventArgs or UnhandledExceptionEventArgs

AndisGrossteins committed:
Reduce excessive log messages from PluginRegistry

Moved _logger.Info call into if statement where it should be

Hirogen commented on src/LogExpert/Program.cs:
```diff
@@ -319,7 +319,7 @@ private static void ShowUnhandledException (object exceptionObject)
     private static void Application_ThreadException (object sender, ThreadExceptionEventArgs e)
     {
-        _logger.Fatal(e);
+        _logger.Fatal(e.Exception, "Application.ThreadException: ");
```
not sure why the exception is needed here?
Could you explain why the exception needs to be logged as this explicitly catches Thread Exceptions?

AndisGrossteins replied:
It originally logged `ThreadExceptionEventArgs` which makes less sense than logging the actual exception and stack trace.
An example from log file:
```
2025-10-02 00:49:09.9159|FATAL|Program|System.Threading.ThreadExceptionEventArgs
```

Hirogen replied:
The Exception will be displayed to the User in the ShowUnhandledException Function anyway, so it's not necessary to explicitly log it.
The user needs then to decide if he posts the exception to the github repo or not.

I Understand where it comes from, but I rather not log every exception there is, but handle them correctly

Hirogen commented on src/LogExpert.UI/Dialogs/RegexHelperDialog.cs:
```diff
@@ -109,12 +109,12 @@ private void OnCaseSensitiveCheckBoxCheckedChanged (object sender, EventArgs e)
         text = comboBoxTestText.Text;
-        comboBoxTestText.Items.Remove(text);
-        comboBoxTestText.Items.Insert(0, text);
+        _ = _testtextHistoryList.Remove(text);
```
could you post a screenshot, in High DPI (150%) and Normal DPI (100%), and also how it looks in the UI?

AndisGrossteins replied:
Those line don't touch anything in GUI, just prevents the `System.ArgumentException` when trying to modify `comboBox.Items` collection with DataSource binding.

Hirogen commented on src/LogExpert/Program.cs:
```diff
@@ -333,7 +333,7 @@ private static void Application_ThreadException (object sender, ThreadExceptionE
     private static void CurrentDomain_UnhandledException (object sender, UnhandledExceptionEventArgs e)
     {
-        _logger.Fatal(e);
+        _logger.Fatal(e.ExceptionObject as Exception, "AppDomain.CurrentDomain.UnhandledException: ");
```
not sure why the exception is needed here?
Could you explain why the exception needs to be logged as this explicitly catches Unhandled Exceptions?

AndisGrossteins replied:
I thought the intent was to log exception no the event arguments.

Hirogen commented on src/PluginRegistry/PluginRegistry.cs:
```diff
@@ -155,10 +155,10 @@ private void LoadPluginAssembly (string dllName, string interfaceName)
-
             if (type.GetInterfaces().Any(i => i.FullName == interfaceName))
             {
+                _logger.Info($"Type {type.FullName} in assembly {assembly.FullName} implements {interfaceName}");
```
I'm not sure whats the benefit here, because if we don't have a plugin of that type of the given interface, we don't know that it's missing, so it seems information will be lost

AndisGrossteins replied:
Then the log messages don't make much sense.

For example, a sample from more than 4000 similar lines in logfile:
```log
2025-10-02 11:28:14.2176|INFO|LogExpert.PluginRegistry.PluginRegistry|Type Microsoft.CodeAnalysis.EmbeddedAttribute in assembly BouncyCastle.Cryptography, Version=2.0.0.0, Culture=neutral, PublicKeyToken=072edcf4a5328938 implements LogExpert.ILogLineColumnizer
2025-10-02 11:28:14.2176|INFO|LogExpert.PluginRegistry.PluginRegistry|Type System.Runtime.CompilerServices.NullableAttribute in assembly BouncyCastle.Cryptography, Version=2.0.0.0, Culture=neutral, PublicKeyToken=072edcf4a5328938 implements LogExpert.ILogLineColumnizer
2025-10-02 11:28:14.2176|INFO|LogExpert.PluginRegistry.PluginRegistry|Type System.Runtime.CompilerServices.NullableContextAttribute in assembly BouncyCastle.Cryptography, Version=2.0.0.0, Culture=neutral, PublicKeyToken=072edcf4a5328938 implements LogExpert.ILogLineColumnizer
2025-10-02 11:28:14.2176|INFO|LogExpert.PluginRegistry.PluginRegistry|Type ThisAssembly in assembly BouncyCastle.Cryptography, Version=2.0.0.0, Culture=neutral, PublicKeyToken=072edcf4a5328938 implements LogExpert.ILogLineColumnizer
```
none of those types actually implement `LogExpert.ILogLineColumnizer`.

Hirogen replied:
yes, but the problem lies in the calling method rather thats where it originates from and the question is, why are those assemblies in the plugin folder? 

Because only the files in the plugins folder are Loaded by the PluginRegistry, maybe the build process is wrong or something other is not working.

But the logline is fine where it was.

as you can see in the PluginRegistry.cs 
"var pluginDir = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "plugins");"

So maybe something else is wrong, but those files should not be in the plugins folder!

AndisGrossteins replied:
There are multiple other dll files besides plugin assemblies in the `plugins` directory.

The point is log messages reported all assemblies as "implements LogExpert.ILogLineColumnizer" even those that don't implement the interface.

AndisGrossteins committed:
Replace buttonRegexHelper with linkLabelRegexHelper

Replaced the `Button` control `buttonRegexHelper` with a `LinkLabel` control `linkLabelRegexHelper` to improve the UI for the search dialog. Updated `TabIndex` values across multiple controls to ensure proper tab navigation order. Adjusted properties of `linkLabelRegexHelper` for accessibility and alignment with the new design.

Modified `groupBoxOptions` and `groupBoxDirection` controls to reflect the new layout, including size and `TabIndex` adjustments. Removed the declaration of `buttonRegexHelper` and added `linkLabelRegexHelper` in the private fields section.

AndisGrossteins committed:
Revert "Log actual exception instead of ThreapdExceptionEventArgs or UnhandledExceptionEventArgs"

This reverts commit 79d59d94ac222cc3ae8a64adaaacfe734afb0356.

Hirogen commented on src/LogExpert.UI/Dialogs/SearchDialog.Designer.cs:
```diff
@@ -20,227 +20,229 @@ protected override void Dispose(bool disposing)
+        labelSearchFor = new Label();
+        checkBoxCaseSensitive = new CheckBox();
+        checkBoxRegex = new CheckBox();
+        linkLabelRegexHelper = new LinkLabel();
```
Why change it from button to LinkLabel?

AndisGrossteins replied:
To align the control group boxes and also to avoid issues with size of the button and text overflow.
Testing on a high-DPI display was breaking the layout and using LinkLabel seems like a better option for that feature.
I've certainly seen this pattern used before for help in a search dialog.

Hirogen commented:
some of the changes were already added, could you please check the latest version, if everything you need has been implemented, and create a new issue if not?
